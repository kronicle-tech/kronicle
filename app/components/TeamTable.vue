<template>
  <table v-if="teams && teams.length > 0" class="table table-dark">
    <thead>
      <tr>
        <th class="team-id">Id</th>
        <th class="team-name">Name</th>
        <th class="team-email-address">Email Address</th>
        <th class="team-area-id">Area Id</th>
        <th class="team-description">Description</th>
        <th class="team-links">Links</th>
      </tr>
    </thead>
    <tbody>
      <tr v-for="(team, teamIndex) in sortedTeams" :key="teamIndex">
        <td class="team-id">{{ team.id }}</td>
        <td class="team-name table-primary">
          <TeamName :team="team" />
        </td>
        <td class="team-email-address">{{ team.emailAddress }}</td>
        <td class="team-area-id">{{ team.areaId }}</td>
        <td class="team-description">
          <Markdown :markdown="team.description" />
        </td>
        <td class="team-links">
          <Links :links="team.links" />
        </td>
      </tr>
    </tbody>
  </table>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { Team } from '~/types/kronicle-service'
import Links from '~/components/Links.vue'
import Markdown from '~/components/Markdown.vue'
import TeamName from '~/components/TeamName.vue'
import { compareTeams } from '~/src/teamComparator'

export default Vue.extend({
  components: {
    Links,
    Markdown,
    TeamName,
  },
  props: {
    teams: {
      type: Array as PropType<Team[]>,
      default: undefined,
    },
  },
  computed: {
    sortedTeams() {
      return [...this.teams].sort(compareTeams)
    },
  },
})
</script>
