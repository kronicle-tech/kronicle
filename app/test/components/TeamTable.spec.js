import { mount } from '@vue/test-utils'
import TeamTable from '@/components/ComponentTable.vue'
import { createTeam } from '~/test/testDataUtils'

describe('TeamTable', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(TeamTable, { propsData })
  }

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when teams prop is set to an empty array', () => {
    beforeEach(() => {
      propsData = {
        teams: [],
      }
    })

    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
    })
  })

  describe('when teams is set to multiple teams', () => {
    beforeEach(() => {
      propsData = {
        teams: [createTeam({ teamNumber: 1 }), createTeam({ teamNumber: 2 })],
      }
    })

    test('renders the teams', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when a team has no description', () => {
    beforeEach(() => {
      propsData = {
        teams: [createTeam(1), createTeam(2)],
      }
      delete propsData.teams[0].description
    })

    test('renders the team without a description', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
