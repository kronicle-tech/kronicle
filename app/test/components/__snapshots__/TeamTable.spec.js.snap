// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamTable when a team has no description renders the team without a description 1`] = `""`;

exports[`TeamTable when teams is set to multiple teams renders the teams 1`] = `""`;
